src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1554,11 +1559,78 @@ AddNodeMetadata(char *nodeName, int32 nodePort,
 
 /*
  * SetWorkerColumn function sets the column with the specified index
+ * on the worker in pg_dist_node, by calling SetWorkerColumnLocalOnly.
+ * It also sends the same command for node update to other metadata nodes.
+ * If anything fails during the transaction, we rollback it.
+ * Returns the new worker node after the modification.
+ */
+WorkerNode *
+SetWorkerColumn(WorkerNode *workerNode, int columnIndex, Datum value)
+{
+	workerNode = SetWorkerColumnLocalOnly(workerNode, columnIndex, value);
+
+	char *metadataSyncCommand = GetMetadataSyncCommandToSetNodeColumn(workerNode,
+																	  columnIndex,
+																	  value);
+
+	SendCommandToWorkersWithMetadata(metadataSyncCommand);
+
+	return workerNode;
+}
+
+
+/*
+ * SetWorkerColumnOptional function sets the column with the specified index
+ * on the worker in pg_dist_node, by calling SetWorkerColumnLocalOnly.
+ * It also sends the same command optionally for node update to other metadata nodes,
+ * meaning that failures are ignored. Returns the new worker node after the modification.
+ */
+WorkerNode *
+SetWorkerColumnOptional(WorkerNode *workerNode, int columnIndex, Datum value)
+{
+	char *metadataSyncCommand = GetMetadataSyncCommandToSetNodeColumn(workerNode,
+																	  columnIndex,
+																	  value);
+
+	List *workerNodeList = TargetWorkerSetNodeList(NON_COORDINATOR_METADATA_NODES,
+												   ShareLock);
+
+	/* open connections in parallel */
+	WorkerNode *worker = NULL;
+	foreach_ptr(worker, workerNodeList)
+	{
+		bool success = SendOptionalCommandListToWorkerInCoordinatedTransaction(
+			worker->workerName, worker->workerPort,
+			CurrentUserName(),
+			list_make1(metadataSyncCommand));
+
+		if (!success)
+		{
+			/* metadata out of sync, mark the worker as not synced */
```

Comment:
should we WARN the user about the situation?

Reply:
The message is kind of hard to understand. Maybe:

Updating the metadata of the node %s:%d is failed on node %s:%d. Metadata on %s:%d is marked as out of sync %( workerNode, node, node)